In [1]:
import cv2
import numpy as np

# How to read image in opencv

In [2]:
#load the image using 'imread' function
img = cv2.imread('my.JPEG')

#To display our image we use 'imshow'
#The first parameter will be title shown on image window
#The second parameter is the image variable
cv2.imshow('Hello World', img)


#'waitKey' allows us to input information when a image window is open
#By leaving it blank it just waits for anykey to be pressed before
#continuing, By placing numbers (except 0), we can specify a delay for
#how long you keep the window open (time is in milliseconds here)

cv2.waitKey(0) #you can leave it empty zero and empty leave thing

#This closes all open windows
#Failure to place this will cause your program to hang
cv2.destroyAllWindows()

#### ^Our picture is cropped
By this higher dimension's picture will not be shown to the screen fully they cropped automatically
because our computer screen is less in dimension and our picture is in higher dimensions like more width and hight
so we can fix this by adjusting the size of our input picture

## image in the form of array

In [4]:
print(img)

[[[141 151 151]
  [141 151 151]
  [142 152 152]
  ...
  [135 144 153]
  [127 136 146]
  [133 142 152]]

 [[142 152 152]
  [141 151 151]
  [141 151 151]
  ...
  [139 148 157]
  [132 141 151]
  [138 147 157]]

 [[140 150 150]
  [139 149 149]
  [141 151 151]
  ...
  [140 149 158]
  [134 143 153]
  [137 145 158]]

 ...

 [[200 139  95]
  [201 140  96]
  [193 132  88]
  ...
  [112 134 199]
  [108 128 193]
  [118 135 202]]

 [[206 143  99]
  [200 137  93]
  [203 140  96]
  ...
  [123 142 209]
  [119 135 204]
  [119 134 203]]

 [[202 140  94]
  [198 135  91]
  [199 136  92]
  ...
  [123 142 210]
  [129 145 214]
  [121 135 207]]]


## shape:- 
We can also check the shape of image by this

In [11]:
img.shape

(5184, 3456, 3)

### ^in this first(5184) is height, 2nd(3456) is width and third(3) is RGB means color

# cv2.imwrite('path', img)

#### How do we save images we edit in OpenCV?

In [17]:
#Simply use 'imwrite' specificing the file name and the image to be saved
cv2.imwrite('output.png', img)
cv2.imwrite('output.jpg', img)

True

In [16]:
img2 = cv2.imread('output.png')
cv2.imshow('output image', img2)
cv2.waitKey() #it waits untill any key to be pressed
cv2.destroyAllWindows()